x-pack/plugins/endpoint/server/services/endpoint/endpoint_query_builders.ts

```diff
@@ -65,3 +67,12 @@ async function getPagingProperties(
     pageIndex: pagingProperties.page_index || config.endpointResultListDefaultFirstPageIndex,
   };
 }
+
+function buildQueryBody(request: KibanaRequest<any, any, any>): Record<string, any> {
+  if (request?.body?.filters && typeof request.body.filters === 'string') {
+    return toElasticsearchQuery(fromKueryExpression(request.body.filters));
```

Comment:
is this the KQL translation magic here?

Reply:
It is using function from the AST (Abstract Syntax Tree) Module to transform the text to an elastic query. Not magical, just classic language syntax tree transformation.

Reply:
I'm going to be using this function too, for the alert list page... let's think about pulling this out into a common location?